packages/input-manager/src/devices/device.ts

```diff
@@ -46,4 +59,33 @@ export abstract class Device extends EventEmitter<DeviceEvents> {
 	async destroy(): Promise<void> {
 		this.removeAllListeners()
 	}
+
+	protected addTriggerEvent(triggerEvent: TriggerEvent): void {
+		this.#triggerEvents.push(triggerEvent)
+		this.emit('trigger')
+	}
+	protected updateTriggerAnalog<T extends TriggerEventArguments>(
+		triggerEvent: Omit<TriggerEvent, 'arguments'>,
+		updateArgumnets: (triggerAnalog: T | undefined) => T
+	): void {
+		const existingIndex = this.#triggerEvents.findIndex((t) => t.triggerId === triggerEvent.triggerId)
+		const trigger: TriggerEvent = existingIndex !== -1 ? this.#triggerEvents[existingIndex] : triggerEvent
+
+		// Update the analog value:
+		trigger.arguments = updateArgumnets(trigger.arguments as T | undefined)
+
+		// Move the trigger to end:
+		this.#triggerEvents.splice(existingIndex, 1)
+		this.#triggerEvents.push(trigger)
+
+		this.emit('trigger')
+	}
+
+	/**
+	 * Returns the next trigger to send to Core.
+	 * If there are no more triggers to send, return undefined
+	 */
+	getNextTrigger(): TriggerEvent | undefined {
```

Comment:
I'm wondering if it would be useful to implement this as a [generator function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator)?

Reply:
I suppose it might be useful, however since we generally rarely use the generator-pattern, I'm of the opinion that it adds more confusion than helps.